providers/implementations/keymgmt/hybrid_kmgmt.c

```diff
@@ -0,0 +1,739 @@
+            OSSL_PARAM_set_all_unmodified(prms);                            \
+            if (!type ## _get_params(s[i], prms))                           \
+                return 0;                                                   \
+            if (OSSL_PARAM_modified(prms)) {                                \
+                if (got) {                                                  \
+                    ERR_raise(ERR_LIB_PROV,                                 \
+                              PROV_R_DUPLICATE_VALUES_FOR_GETTER);          \
+                    return 0;                                               \
+                }                                                           \
+                got = 1;                                                    \
+                *param = *prms;                                             \
+            }                                                               \
+        }                                                                   \
+        return 1;                                                           \
+    }
+
+GET_SUBALG_PARAM(key, EVP_PKEY)
+GET_SUBALG_PARAM(ctx, EVP_PKEY_CTX)
+
+#define SET_SUBALG_PARAM(lname, type)                                       \
+    static int lname ## _subalg_set_param                                   \
+        (unsigned int n, type **s, const OSSL_PARAM *param)                 \
+    {                                                                       \
+        OSSL_PARAM prms[2] = { OSSL_PARAM_END, OSSL_PARAM_END };            \
+        unsigned int i;                                                     \
+                                                                            \
+        *prms = *param;                                                     \
+        for (i = 0; i < n; i++) {                                           \
+            if (!type ## _set_params(s[i], prms))                           \
+                return 0;                                                   \
+        }                                                                   \
+        return 1;                                                           \
+    }
+
+SET_SUBALG_PARAM(key, EVP_PKEY)
+SET_SUBALG_PARAM(ctx, EVP_PKEY_CTX)
+
+int ossl_hybrid_kmgmt_get_params(void *vkey, OSSL_PARAM params[])
+{
+    HYBRID_PKEY *key = (HYBRID_PKEY *)vkey;
+    OSSL_PARAM prms[2] = { OSSL_PARAM_END, OSSL_PARAM_END };
+    OSSL_PARAM *p;
+    unsigned char *q;
+    unsigned int i;
+    int type;
+
+    if (ossl_param_is_empty(params))
+        return 1;
+
+    for (p = params; params->key != NULL; params++) {
+        type = ossl_param_find_pidx(p->key);
+        switch (type) {
+        case PIDX_PKEY_PARAM_SECURITY_BITS:
+            if (!OSSL_PARAM_set_size_t(p, key->security_bits))
+                return 0;
+            break;
+
+        case PIDX_PKEY_PARAM_BITS:
+            if (!!OSSL_PARAM_set_size_t(p, key->bits))
```

Comment:
`!!` ??

Reply:
Typo, will fix.  Guess this one isn't covered well by our tests :disappointed: